membership/application_test.go

```diff
@@ -0,0 +1,135 @@
+package membership
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestCreateMembership(t *testing.T) {
+	t.Run("멤버십을 생성한다.", func(t *testing.T) {
+		app := NewApplication(*NewRepository(map[string]Membership{}))
+		req := CreateRequest{"jenny", "naver"}
+		res, err := app.Create(req)
+		assert.Nil(t, err)
+		assert.NotEmpty(t, res.ID)
+		assert.Equal(t, req.MembershipType, res.MembershipType)
+	})
+
+	t.Run("이미 등록된 사용자 이름이 존재할 경우 실패한다.", func(t *testing.T) {
+		app := NewApplication(*NewRepository(map[string]Membership{}))
+		req := CreateRequest{"jenny", "naver"}
+		_, _ = app.Create(req)
+		req = CreateRequest{"jenny", "payco"}
+		_, err := app.Create(req)
+		assert.Equal(t, fmt.Errorf("same_name"), err)
+	})
+
+	t.Run("사용자 이름을 입력하지 않은 경우 실패한다.", func(t *testing.T) {
+		app := NewApplication(*NewRepository(map[string]Membership{}))
+		req := CreateRequest{"", "naver"}
+		_, err := app.Create(req)
+		assert.Equal(t, fmt.Errorf("no_name"), err)
+	})
+
+	t.Run("멤버십 타입을 입력하지 않은 경우 실패한다.", func(t *testing.T) {
+		app := NewApplication(*NewRepository(map[string]Membership{}))
+		req := CreateRequest{"jenny", ""}
+		_, err := app.Create(req)
+		assert.Equal(t, fmt.Errorf("no_membership"), err)
+	})
+
+	t.Run("naver/toss/payco 이외의 타입을 입력한 경우 실패한다.", func(t *testing.T) {
+		app := NewApplication(*NewRepository(map[string]Membership{}))
+		req := CreateRequest{"jenny", "kakao"}
+		_, err := app.Create(req)
+		assert.Equal(t, fmt.Errorf("wrong_membership"), err)
+	})
+}
+
+func TestUpdate(t *testing.T) {
+	t.Run("membership 정보를 갱신한다.", func(t *testing.T) {
+		app := NewApplication(*NewRepository(map[string]Membership{}))
+		creq := CreateRequest{"jenny", "naver"}
+		_, _ = app.Create(creq)
```

Comment:
```suggestion
		app.Create(creq)
```
함수를 호출한 결과값을 사용하지 않는 경우에는 return 값을 굳이 받을 필요가 없습니다